ha/note.c

```diff
@@ -140,6 +142,56 @@ M0_INTERNAL void m0_ha_state_single_post(struct m0_ha_nvec *nvec)
 	m0_ha_msg_nvec_send(nvec, 0, false, M0_HA_NVEC_SET, NULL);
 }
 
+static void m0_ha_add_dynamic_fid_to_confc(
+			struct m0_conf_cache *cache,
+			struct m0_conf_obj   *base_obj,
+			struct m0_ha_note    *nv_note,
+			uint64_t              ignore_same_state)
+{
+	struct m0_conf_obj     *new_obj;
+	struct m0_conf_dir     *dir;
+        enum m0_ha_obj_state    prev_ha_state;
+	struct m0_conf_process *proc1;
+	struct m0_conf_process *proc2;
+	int                     rc;
+
+	M0_LOG(M0_DEBUG, "Received fid: "FID_F" lookedup fid :"FID_F,
+		FID_P(&nv_note->no_id), FID_P(&base_obj->co_id));
+
+	rc = m0_conf_obj_find(cache, &nv_note->no_id, &new_obj);
+	if (rc == 0) {
+		prev_ha_state = base_obj->co_ha_state;
+		new_obj->co_ha_state = nv_note->no_state;
+
+		if (new_obj->co_parent == NULL) {
+			dir = M0_CONF_CAST(base_obj->co_parent, m0_conf_dir);
+			m0_conf_dir_add(dir, new_obj);
+		}
+		proc1 = M0_CONF_CAST(base_obj, m0_conf_process);
+		proc2 = M0_CONF_CAST(new_obj, m0_conf_process);
+
+		if (proc1 != NULL && proc2 != NULL) {
+			if (proc1->pc_endpoint != NULL) {
+				proc2->pc_endpoint = m0_strdup(proc1->pc_endpoint);
+			}
+		}
+		rc = m0_conf_dir_new(new_obj, &M0_CONF_PROCESS_SERVICES_FID,
```

Comment:
if proc2 is null by any chance, then this will assert, so protect under null check